packages/docusaurus/src/commands/serve.ts

```diff
@@ -36,6 +37,14 @@ export async function serve(
   siteDirParam: string = '.',
   cliOptions: Partial<ServeCLIOptions> = {},
 ): Promise<void> {
+  const program = new Command();
+
+  program
+	.option('--no-clean-urls', 'Disable clean URLs (keep .html extensions)')
+	.parse(process.argv);
+
+  const { noCleanUrls } = program.opts();
```

Comment:
CI reports error:

> Error:   46:11  error  'noCleanUrls' is assigned a value but never used  @typescript-eslint/no-unused-vars


You have to actually use the option for the PR to have any meaningful effect